src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,165 @@
+    item: Section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * render custom title section
+   */
+  renderSectionTitle?: (item: Section, index: number, expanded: boolean) => React.ReactNode
+  /**
+   * render custom content
+   */
+  renderContent?: (
+    item: Section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * accordion title
+   */
+  title?: string
+  /**
+   * accordion container style
+   */
+  itemContainerStyle?: ViewStyleProp
+  /**
+   * accordion title style
+   */
+  titleStyle?: TextTitleStyleProp
+  /**
+   * accordion header style
+   */
+  headerContainerStyle?: ViewStyleProp
+  /**
+   * accordion body style
+   */
+  contentContainerStyle?: ViewStyleProp
+  /**
+   * animation type when open
+   * default: easeInEaseOut
+   */
+  openAnimation?: AnimationType
+  /**
+   * animation type when close
+   * default: easeInEaseOut
+   */
+  closeAnimation?: AnimationType
+  /**
+   * duration when open
+   * default: 300
+   */
+  openDuration?: number
+  /**
+   * duration when close
+   * default: 300
+   */
+  closeDuration?: number
+}
+
+export interface AccordionProps extends CommonAccordionProps {
+  /**
+   * section data sets
+   */
+  sections: Section[]
+  /**
+   * keyExtractor for Flatlist
+   */
+  keyExtractor?: (item: Section, index: number) => string
+  /**
+   * enable expand multiple item
+   * default: false
+   */
+  expandMultiple?: boolean
+  /**
+   * accordion wrapper style
+   */
+  wrapperStyle?: ViewStyleProp
+}
+
+const Accordion = React.forwardRef<FlatList, AccordionProps>(
+  ({sections, expandMultiple = false, keyExtractor, wrapperStyle, ...props}, ref) => {
+    const [array, setArray] = useState<string[]>([])
+
+    const _keyExtractor = useMemo(
+      () => keyExtractor || ((_: Section, index: number) => index.toString()),
+      [keyExtractor],
+    )
+
+    const onPress = useCallback(
+      (key: string) => {
+        const index = array.findIndex(e => e === key)
+        let newArray = [...array]
+
+        if (expandMultiple) {
+          if (index >= 0) {
+            newArray.splice(index, 1)
+          } else {
+            newArray.push(key)
+          }
+        } else {
+          newArray = index >= 0 ? [] : [key]
+        }
+
+        setArray(newArray)
+      },
+      [array, expandMultiple],
+    )
+
+    const renderItem = useCallback(
+      ({item, index}: {item: Section; index: number}) => (
+        <AccordionItem
+          key={_keyExtractor(item, index)}
+          title={item?.title}
+          expanded={array.includes(_keyExtractor(item, index))}
+          {...props}
+          item={item}
+          keyExtractorItem={_keyExtractor(item, index)}
+          onPress={onPress}
+          index={index}>
+          <Text>{item?.content ?? ''}</Text>
+        </AccordionItem>
+      ),
+      [_keyExtractor, onPress, array, props],
```

Comment:
Have you checked the `renderItem` re-render because you pass `array` to the dependency?

Reply:
I believe we need the array there because the expanded prop depends on the new array. We could create a new function to check should expanded or not, however we still need to array for dependency